File: Graduater/Core/Entities/Group.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace Core.Entities;

public partial class Group
{
    public string Name { get; set; } = null!;

    public string Description { get; set; } = null!;

    public int CreatorUserId { get; set; }

    public virtual ICollection<Assignment> Assignments { get; set; } = new List<Assignment>();

    [ForeignKey(nameof(CreatorUserId))]
    public virtual User CreatorUser { get; set; } = new();

    public virtual ICollection<GroupUser> GroupUsers { get; set; } = new List<GroupUser>();
}

File: models/Group.ts
import DataBaseEntity from "./DataBaseEntity";
import Assignment from "./Assignment";
import GroupUser from "./GroupUser";
import User from "./User";

export default interface Group extends DataBaseEntity{
    assignments?: Assignment[];
    creatorUser?: User;
    creatorUserId: number;
    description: string;
    groupUsers?: GroupUser[];
    name: string;
}